xc/xenops_server_xen.ml

```diff
@@ -2795,14 +2842,16 @@ module VIF = struct
 
   let get_device_action_request vm vif =
     with_xc_and_xs
-      (fun xc xs ->
-         try
-           let (device: Device_common.device) = device_by_id xc xs vm Device_common.Vif Newest (id_of vif) in
-           if Hotplug.device_is_online ~xs device
-           then None
-           else Some Needs_unplug
-         with Device_not_connected ->
-           None
+      (fun xc xs -> match vif.backend with
+        | Network.Sriov _ -> None
```

Comment:
I'm not convinced that we never need any cleanup here. See comment on unplug.

Reply:
Responded in comment above.